packages/marketplace-management/src/components/ui/users/edit-user.tsx

```diff
@@ -45,13 +43,35 @@ export const formFields: Record<FieldType, FormFieldInfo> = {
   },
 }
 
+export const sortAddRemoveGroups = (editingUser: UserModel, groupIds: string[]) => {
+  const currentGroups = editingUser.groups ?? []
+  const removeIds = currentGroups.filter((group) => !groupIds.find((groupId) => groupId === group))
```

Comment:
use includes instead of reduce foreach
```suggestion
  const removeIds = currentGroups.filter((group) => !groupIds.includes(group))
```